src/heretic/model.py

```diff
@@ -202,11 +208,14 @@ def abliterate(
                     layer_refusal_direction,
                     layer_refusal_direction,
                 ).to(self.model.dtype)
-
+                
                 for matrix in matrices:
                     # In-place subtraction is safe as we're not using Autograd.
-                    matrix.sub_(weight * (projector @ matrix))
-
+                    # Ensure projector is on the same device as the matrix for multi-GPU support.
+                    device_projector = projector.to(matrix.device)
+                    matrix.sub_(weight * (device_projector @ matrix))
+                    # Delete device-specific projector immediately to free memory
+                    del device_projector
```

Comment:
This also shouldn't be necessary, because either we loop and immediately overwrite `device_projector`, releasing the previous reference just like `del` does, or the method returns and `device_projector` goes out of scope, also releasing the reference.

Note that in Python, `del x` does **not** call `x.__del__()`. The `__del__` method (which libraries like PyTorch use to clean up resources) is only called on object destruction, and the GC doesn't care whether a reference is released with `del` or by overwriting it with another value. If you want to make sure memory is released immediately, you have to follow up `del` with `gc.collect()`. But since you got this working without it and `gc.collect()` is already called on each trial anyway, there's probably no need for this line at all.